packages/maskbook/src/plugins/Ethereum/UI/GasNowDialog.tsx

```diff
@@ -0,0 +1,229 @@
+                type: 'standard',
+                gasPrice: gasNow?.standard ?? 0,
+                time: t('plugin_gas_now_dialog_min', { time: 3 }),
+            },
+            {
+                title: t('plugin_gas_now_dialog_fast'),
+                type: 'fast',
+                gasPrice: gasNow?.fast ?? 0,
+                time: t('plugin_gas_now_dialog_min', { time: 1 }),
+            },
+            {
+                title: t('plugin_gas_now_dialog_custom'),
+                type: 'custom',
+                gasPrice: customGasToWei.toNumber(),
+                time: t('plugin_gas_now_dialog_sec', { time: 15 }),
+            },
+        ],
+        [gasNow, customGas],
+    )
+
+    const validationMessage = useMemo(() => {
+        if (select === 2 && customGasToWei.isEqualTo(0)) return t('plugin_gas_now_dialog_zero_gas')
+        return ''
+    }, [customGasToWei, options, select])
+
+    const onConfrim = useCallback(() => {
+        currentGasPriceSettings.value = select == 2 ? customGasToWei.toNumber() : options[select].gasPrice
+        const type = options[select].type as keyof GasNow
+        setDialog({ open: false, type })
+    }, [customGasToWei, options, select, setDialog])
+
+    return (
+        <InjectedDialog
+            open={open}
+            onClose={closeDialog}
+            title={t('plugin_gas_now_dialog_title')}
+            DialogProps={{ maxWidth: 'xl' }}>
+            <DialogContent className={classes.content}>
+                <List className={classes.list}>
+                    {options.map((option, i) => (
+                        <ListItem
+                            onClick={() => (i === 2 || gasNow) && setSelect(i)}
+                            className={classNames(
+                                classes.box,
+                                i === select ? classes.boxSelected : undefined,
+                                i !== 2 && !gasNow ? classes.disableBox : undefined,
+                            )}
+                            key={i}>
+                            <div className={classes.head}>
+                                <Typography className={classes.title}>{option.title}</Typography>
+                                <FiberManualRecordIcon
+                                    className={classNames(classes.dot, i === select ? classes.dotSelector : undefined)}
+                                />
+                            </div>
+                            {gasNow || i === 2 ? (
+                                <>
+                                    <div>
+                                        {i == 2 ? (
+                                            <div className={classes.gweiBox}>
+                                                <TextField
+                                                    className={classes.customInput}
+                                                    value={customGas}
+                                                    onChange={(e) => {
```

Comment:
```suggestion
                                                    value={customGas}
                                                    InputProps={{
                                                        inputMode: 'decimal',
                                                        pattern: '^[0-9]*[.,]?[0-9]*$',
                                                    }}
                                                    onChange={(e) => {
```

Reply:
'pattern' does not exist in type 'Partial<InputProps> | Partial<FilledInputProps> | Partial<OutlinedInputProps>'